openedx/adg/common/mailchimp_pipeline/tests/test_tasks.py

```diff
@@ -2,34 +2,110 @@
+from openedx.core.djangolib.testing.utils import skip_unless_lms
+from student.models import UserProfile
 from student.tests.factories import UserFactory
 
 
 @pytest.mark.django_db
 @pytest.mark.mute_signals
-def test_task_send_user_info_to_mailchimp(mocker):
+def test_task_send_user_info_to_mailchimp_user_created(mocker):
     """
-    Assert that mailchimp client is called with valid data for syncing user's info
+    Assert that mailchimp client is called with valid data when User is created.
     """
+    mock_mailchimp_client = mocker.patch.object(mailchimp_tasks, 'MailchimpClient', autospec=True)
+    mock_create_or_update_list_member = mock_mailchimp_client().create_or_update_list_member
+
     user = UserFactory()
+    mailchimp_tasks.task_send_user_info_to_mailchimp(User, user)
+
+    user_json = {
+        'email_address': user.email,
+        'status_if_new': 'subscribed',
+        'merge_fields': {
+            'DATEREGIS': str(user.date_joined.strftime('%m/%d/%Y')),
+            'USERNAME': user.username
+        },
+    }
+    mock_create_or_update_list_member.assert_called_once_with(email=user.email, data=user_json)
+
+
+@pytest.mark.django_db
+@pytest.mark.mute_signals
+def test_task_send_user_info_to_mailchimp_profile_created(mocker):
+    """
+    Assert that mailchimp client is called with valid data when UserProfile is created.
+    """
     mock_mailchimp_client = mocker.patch.object(mailchimp_tasks, 'MailchimpClient', autospec=True)
     mock_create_or_update_list_member = mock_mailchimp_client().create_or_update_list_member
-    mailchimp_tasks.task_send_user_info_to_mailchimp(user)
+
+    profile = UserFactory().profile
+    mailchimp_tasks.task_send_user_info_to_mailchimp(UserProfile, profile)
 
     user_json = {
-        "email_address": user.email,
-        "merge_fields": {
-            "DATEREGIS": str(user.date_joined.strftime("%m/%d/%Y")),
-            "FULLNAME": user.get_full_name(),
-            "USERNAME": user.username
+        'email_address': profile.user.email,  # pylint: disable=no-member
+        'status_if_new': 'subscribed',
+        'merge_fields': {
+            'LOCATION': profile.city,  # pylint: disable=no-member
+            'FULLNAME': profile.name
         },
-        "status_if_new": "subscribed",
     }
+    mock_create_or_update_list_member.assert_called_once_with(email=profile.user.email, data=user_json)  # pylint: disable=no-member
 
-    mock_create_or_update_list_member.assert_called_once_with(email=user.email, data=user_json)
+
+@pytest.mark.django_db
+@pytest.mark.mute_signals
+@skip_unless_lms
+def test_task_send_user_info_to_mailchimp_user_application_created(mocker):
+    """
+    Assert that mailchimp client is called with valid data when User is created.
+    """
+    mock_mailchimp_client = mocker.patch.object(mailchimp_tasks, 'MailchimpClient', autospec=True)
+    mock_create_or_update_list_member = mock_mailchimp_client().create_or_update_list_member
```

Comment:
These two lines are repeated in all tests let's please define a fixture and use it in all these tests.